test/smoke/dashCore.js

```diff
@@ -1,18 +1,116 @@
-describe('DashCore', () => {
-  describe('All nodes', () => {
-    // `subversion` from `getNetworkInfo` response.
-    it('should have correct network type');
-
-    // The number of blocks should be almost the same (-3/+3) and block hash of particular
-    // block height should be the same. Using `blocks` and `bestblockhash` from `GetBlockChainInfo`.
-    it('should propagate blocks');
+const RpcClient = require('@dashevo/dashd-rpc/promise');
+const SSH = require('simple-ssh');
+const fs = require('fs');
+
+const getNetworkConfig = require('../../lib/test/getNetworkConfig');
+
+const networkConfig = getNetworkConfig();
+
+
+describe('All nodes', () => {
+  networkConfig.inventory.masternodes.hosts.forEach((nodeName) => {
+    describe(`DashCore ${nodeName}`, () => {
+      const config = {
+        protocol: 'http',
+        user: 'dashrpc',
+        pass: 'password',
+        host: networkConfig.inventory._meta.hostvars[nodeName].public_ip,
+        port: 20002,
+      };
+      const rpc = new RpcClient(config);
+
+      it('should have correct network type', async () => {
+        const blockHash = await rpc.getNetworkInfo();
+        const masternodelist = await rpc.masternodelist();
+        const version = masternodelist.result[Object.keys(masternodelist.result)[0]].daemonversion;
+        expect(blockHash.error).to.be.equal(null);
+        expect(blockHash.result.networkactive).to.be.equal(true);
+        expect(blockHash.result.subversion).to.be.equal(`/Dash Core:${version}(${networkConfig.network.type}=${networkConfig.network.name})/`);
+      });
+    });
   });
-  describe('Masternodes', () => {
-    // masternode status
-    it('should be masternodes');
+
+  // The number of blocks should be almost the same (-3/+3) and block hash of particular
+  // block height should be the same. Using `blocks`
+  // and `bestblockhash` from `GetBlockChainInfo`.
+  it('should have correct blockhash and blocks count', async () => {
+    const hosts = networkConfig.inventory.masternodes.hosts;
+    let blockCounts = [];
+    for (let i = 0; i < hosts.length; i++) {
+      const config = {
+        protocol: 'http',
+        user: 'dashrpc',
+        pass: 'password',
+        host: networkConfig.inventory._meta.hostvars[hosts[i]].public_ip,
+        port: 20002,
+      };
+      const rpc = new RpcClient(config);
+
+      const bestBlockHash = await rpc.getBestBlockHash();
+      const blockCount = await rpc.getBlockCount();
```

Comment:
I think here `getBlockchainInfo` will be enough. We don't need to compare it with `getBestBlockHash` and  `getBlockCount` with `getBlockchainInfo`. They are always will be equal.

But we need to check hashes on particular height. So In addition to `blockCounts` we need to aggregate block hashes by heights and make sure that they are equal. For example:
```
if (!blockhashes[blockchainInfo.result.blocks]) {
   blockhashes[blockchainInfo.result.blocks] = blockchainInfo.result.bestblockhash;
}
expect(blockhashes[blockchainInfo.result.blocks]).to.be.equal(blockchainInfo.result.bestblockhash);
```